Signup page

udenizdemirbilek opened the pull request:
#19 Draft commit for error checking
Signup Form is functional, creates a user in [firebase](https://console.firebase.google.com/project/capstone-turkey-giftify/authentication/users), 
![image](https://user-images.githubusercontent.com/81304552/130632125-fcfadabb-7233-4b28-a1bf-f9ef2c8c28ea.png)
stores the information in the [database](https://console.firebase.google.com/project/capstone-turkey-giftify/firestore/data/~2Fusers)
![image](https://user-images.githubusercontent.com/81304552/130632276-7f70cbd8-fcbb-48a4-827c-c651ba2b7b26.png)

Also google sign-in is also working, please see first image.

Desktop 
![image](https://user-images.githubusercontent.com/81304552/130632862-d4d4dd94-e1e9-4953-abbf-9ccdfcc8bdb7.png)
Tablet
![image](https://user-images.githubusercontent.com/81304552/130633187-b13b939a-cd97-4143-a953-4ea960497cb8.png)
Mobile
![image](https://user-images.githubusercontent.com/81304552/130633088-1661d532-e2a5-4b13-8ba2-451b392f857d.png)

udenizdemirbilek committed:
draft signup

udenizdemirbilek committed:
draft signup

udenizdemirbilek committed:
draft sign-up

udenizdemirbilek committed:
draft signUp2

udenizdemirbilek committed:
merge signup with old work

udenizdemirbilek committed:
draft signUp page

udenizdemirbilek committed:
draft facebook login

halitbatur commented on .github/workflows/node.js.yml:
```diff
@@ -1,33 +1,33 @@
-
-on:
-  push:
-    branches: [ dev ]
```
there's a lot of files here that should not be changed please check end of lind error again and fix it

halitbatur commented on src/containers/SignUp/SignUp.jsx:
```diff
@@ -0,0 +1,264 @@
+        const { credential } = result;
+        console.log(credential);
+
+        // This gives you a Google Access Token. You can use it to access the Google API.
```
can we have this comment on top of line 79

halitbatur commented on src/containers/SignUp/SignUp.jsx:
```diff
@@ -0,0 +1,264 @@
+      .then((result) => {
+        /** @type {firebase.auth.OAuthCredential} */
+        const { credential } = result;
+        console.log(credential);
```
no console.logs please

halitbatur commented on src/containers/SignUp/SignUp.jsx:
```diff
@@ -0,0 +1,264 @@
+        // Handle Errors here.
+        const errorCode = error.code;
+        const errorMessage = error.message;
+        console.log(errorCode, errorMessage);
```
no console.logs

halitbatur commented on src/containers/SignUp/SignUp.jsx:
```diff
@@ -0,0 +1,264 @@
+
+        // This gives you a Facebook Access Token. You can use it to access the Facebook API.
+        const { accessToken } = credential;
+        console.log(accessToken, user);
```
no console.logs

halitbatur commented on src/containers/SignUp/SignUp.jsx:
```diff
@@ -0,0 +1,264 @@
+        // const {email} = error;
+        // The firebase.auth.AuthCredential type that was used.
+        // const {credential} = error;
+        console.log(errorCode, errorMessage);
```
no console.logs

udenizdemirbilek committed:
Saving files before refreshing line endings

udenizdemirbilek committed:
Normalize all the line endings

ceyhungulbas reviewed (changes requested):
Hey Deniz! Eline saglik. Looks like its really looking good and functionality workin nice but I just found teeny issue about functionality.

![resim](https://user-images.githubusercontent.com/44509662/131211078-a2b7bb85-d466-411a-b6b3-52c4a267a8e4.png)

- As you see I signed up but my mail is not displaying on doc field, is there a way to add this info? Because we are going to use this information sending email between buyer and seller.
- For Re-Enter Password part: I wrote first aha123 then ehe123 then clicked sign up. But got any error message, just refreshing page. If we have a time, maybe we can add this thing too?

udenizdemirbilek commented:
> 
> 
> Hey Deniz! Eline saglik. Looks like its really looking good and functionality workin nice but I just found teeny issue about functionality.
> 
> ![resim](https://user-images.githubusercontent.com/44509662/131211078-a2b7bb85-d466-411a-b6b3-52c4a267a8e4.png)
> 
>     * As you see I signed up but my mail is not displaying on doc field, is there a way to add this info? Because we are going to use this information sending email between buyer and seller.
> 
>     * For Re-Enter Password part: I wrote first aha123 then ehe123 then clicked sign up. But got any error message, just refreshing page. If we have a time, maybe we can add this thing too?

Fixed both issues, also signing up with google and facebook should also create a database user entry with email, name and photoURL now.

Added a new rule to check two passwords to be the same and alerts the user if they are not.

udenizdemirbilek committed:
google and facebook login draft

udenizdemirbilek committed:
google and facebook login2

udenizdemirbilek committed:
Add uid to user db

ceyhungulbas reviewed (approved):
functionality is working very well. Lookin good! Eline saglik Deniz

halitbatur commented on src/containers/SignUp/SignUp.jsx:
```diff
@@ -0,0 +1,258 @@
+          >
+            {cities.map((city, index) => (
+              // eslint-disable-next-line react/no-array-index-key
+              <option key={index}>{city}</option>
```
lets not have the index as a key here maybe it can be the city

udenizdemirbilek replied:
Taken a note in the bug bash file!

udenizdemirbilek committed:
Merge branch 'dev' into signupPage
